pkg/cloud/services/ec2/launchtemplate_test.go

```diff
@@ -100,45 +121,151 @@ func TestGetLaunchTemplate(t *testing.T) {
 						nil,
 					))
 			},
-			check: func(launchtemplate *expinfrav1.AWSLaunchTemplate, userdatahash string, err error) {
+			check: func(launchTemplate *expinfrav1.AWSLaunchTemplate, userDataHash string, err error) {
 				if err != nil {
-					t.Fatalf("did not expect error: %v", err)
+					t.Fatalf("did not expect error, but got: %v", err)
+				}
+
+				if userDataHash != "" {
+					t.Fatalf("Did not expect a userdata hash, but got something: %s", userDataHash)
 				}
 
-				if userdatahash != "" {
-					t.Fatalf("Did not expect a userdata hash, but got something: %s", userdatahash)
+				if launchTemplate != nil {
+					t.Fatalf("Did not expect anything but got something: %+v", launchTemplate)
+				}
+			},
+		},
+		{
+			name:               "internel server error",
+			launchTemplateName: "foo",
+			expect: func(scope *mocks.MockEC2ScopeMockRecorder, m *mock_ec2iface.MockEC2APIMockRecorder) {
+				scope.V(2).Return(klogr.New())
+				m.DescribeLaunchTemplateVersions(gomock.Eq(&ec2.DescribeLaunchTemplateVersionsInput{
+					LaunchTemplateName: aws.String("foo"),
+					Versions:           []*string{aws.String("$Latest")},
+				})).
+					Return(nil, awserrors.NewFailedDependency("dependency failure"))
+			},
+			check: func(launchTemplate *expinfrav1.AWSLaunchTemplate, userDataHash string, err error) {
+				if err == nil {
+					t.Fatalf("expected an error but got nothing")
+				}
+				if userDataHash != "" {
+					t.Fatalf("Did not expect a userdata hash, but got something: %s", userDataHash)
+				}
+				if launchTemplate != nil {
+					t.Fatalf("Did not expect anything but got something: %+v", launchTemplate)
+				}
+			},
+		},
+		{
+			name:               "nil output with no error",
+			launchTemplateName: "foo",
+			expect: func(scope *mocks.MockEC2ScopeMockRecorder, m *mock_ec2iface.MockEC2APIMockRecorder) {
+				scope.V(2).Return(klogr.New())
+				m.DescribeLaunchTemplateVersions(gomock.Eq(&ec2.DescribeLaunchTemplateVersionsInput{
+					LaunchTemplateName: aws.String("foo"),
+					Versions:           []*string{aws.String("$Latest")},
+				})).
+					Return(nil, nil)
+			},
+			check: func(launchTemplate *expinfrav1.AWSLaunchTemplate, userDataHash string, err error) {
+				if err != nil {
+					t.Fatalf("did not expect error: %v", err)
+				}
+				if userDataHash != "" {
+					t.Fatalf("Did not expect a userdata hash, but got something: %s", userDataHash)
+				}
+				if launchTemplate != nil {
+					t.Fatalf("Did not expect anything but got something: %+v", launchTemplate)
+				}
+			},
+		},
+		{
+			name:               "successfully fetched launch template versions",
```

Comment:
In success case, is it possible to add the following scenario:
Should add additionalSecurityGroups in launch template if `DescribeLaunchTemplateVersions` returns output with additionalSecurityGroup.
No need for separate test case, you can add this in existing test `successfully fetched launch template versions`. Wdyt?

Reply:
done